pkg/scaling/cache/scalers_cache.go

```diff
@@ -42,14 +42,23 @@ type ScalersCache struct {
 }
 
 type ScalerBuilder struct {
-	Scaler  scalers.Scaler
-	Factory func() (scalers.Scaler, error)
+	Scaler      scalers.Scaler
+	TriggerName string
```

Comment:
I'm not totally sure about this, the name is part of the scaler IMO. 
Even it's more effort, I feel this value as a scaler property.
WDYT?

Reply:
agree.  It assumed to be a property of "scaler". 
But if we put it inside the scaler,  per my understanding, it requires changes inside the plugged-in scaler .  Per comment in https://github.com/kedacore/keda/issues/3588#issuecomment-1227355883, `we could implement this for all scalers at once.`,  I guess the expectation is to handle it in a common place instead of changes the different scalers' implementation to add this field. 
So,  adding it as a separate field is the most straightforward way.

Reply:
I'd say he meant that using that feature we can unify where we do the addition, but let's "invoke" him xD @zroubalik ?